browser/net/brave_proxying_url_loader_factory.cc

```diff
@@ -610,7 +611,17 @@ void BraveProxyingURLLoaderFactory::InProgressRequest::
 void BraveProxyingURLLoaderFactory::InProgressRequest::OnRequestError(
     const network::URLLoaderCompletionStatus& status) {
   if (!request_completed_) {
-    target_client_->OnComplete(status);
+    // Make a non-const copy of status so that |should_collapse_initiator| can
+    // be modified
+    network::URLLoaderCompletionStatus collapse_status(status);
+
+    if (base::FeatureList::IsEnabled(
+            ::brave_shields::features::kBraveAdblockCollapseBlockedElements) &&
+        ctx_->blocked_by == brave::kAdBlocked) {
+      collapse_status.should_collapse_initiator = true;
+    }
+
+    target_client_->OnComplete(collapse_status);
```

Comment:
can we please add a test?

Reply:
I did look into this briefly - collapsed elements cannot be detected in JS, and the only API Chromium exposes for this is [`HTMLImageElement::IsCollapsed()`](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/html/html_image_element.h;l=140;bpv=0;bpt=1), which is pretty deep in Blink.

Is it allowed to access the Blink elements in browsertests? Otherwise we could probably try some hacky JS where we check the size of a parent element, but it seems fragile.

Reply:
I tried for a bit to access `IsCollapsed`, but I don't see any convenient way to do it. I've used the JS `clientHeight` property instead, which I've found works here.